cmd/2019_7/main.go

```diff
@@ -0,0 +1,62 @@
+package main
+
+import (
+	"flag"
+
+	"github.com/afarbos/aoc/pkg/intcode"
+	"github.com/afarbos/aoc/pkg/mathematic"
+	"github.com/afarbos/aoc/pkg/perm"
+	"github.com/afarbos/aoc/pkg/read"
+	"github.com/afarbos/aoc/pkg/utils"
+)
+
+var flagInput string
+
+func init() {
+	utils.Init(&flagInput)
+}
+
+func findMaxThrusterSignal(instructions []int) int {
+	var (
+		settingsSet = map[int]struct{}{
+			0: {},
+			1: {},
+			2: {},
+			3: {},
+			4: {},
+		}
+		signal = mathematic.MaxInt()
+	)
+
+	perm.Ints(settingsSet, func(settings []int) {
+		var (
+			input             = 0
+			inputs            = make([]int, 2)
+			localInstructions = make([]int, len(instructions))
+		)
+
+		for _, phase := range settings {
+			copy(localInstructions, instructions)
+			inputs[0] = phase
+			inputs[1] = input
+			input = intcode.Compute(localInstructions, intcode.Equals, inputs...)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [line is 81 characters, out of limit 80](https://app.codacy.com/manual/afarbos/aoc/pullRequest?prid=5498712)